resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -199,4 +199,13 @@ private[spark] object KubernetesConf {
       .replaceAll("[^a-z0-9\\-]", "")
       .replaceAll("-+", "-")
   }
+
+  /**
+   * Build a resources name based on the vendor device plugin naming
+   * convention of: vendor-domain/resource. For example, an Nvidia GPU is
+   * advertised as nvidia.com/gpu.
+   */
+  def buildKubernetesResourceName(vendor: String, resourceName: String): String = {
+    s"${vendor}/${resourceName}"
```

Comment:
The Kubernetes resource name uses the format `<vendor>.com/gpu`. Are `vendor` expected to contain the full domain name like `nvidia.com` or just the vendor name?

Reply:
meant to be vendor and domain like the comment says,  I can update the name of the variable if it makes it more clear?

Reply:
What's the reason of requiring the full vendor domain name?

Reply:
that is what k8s takes from reading the docs, if I am wrong please point me to correct docs

Reply:
https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/

Reply:
if we don't allow that then I don't know how we can be generic for any type

Reply:
https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/. It seems from the doc that only the vendor name needs to be configurable.

Reply:
sorry maybe I'm missing it but that doc says:

 limits:
          nvidia.com/gpu: 1 # requesting 1 GPU

nvidia is vendor, .com is domain, resource is gpus.

This PR is trying to handle any generic resource type not just GPUs so the vendor has to have both vendor and domain.  Otherwise we would have to put something in that say for gpu's use .com for domain.

Again I'm just going by https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/, maybe domain is always .com?

Reply:
Can you update the documentation of the two new options to make it clear that they expect vendor domain names?

Reply:
yep I'll update shortly, thanks for looking.

Reply:
actually one question here.  I was leaving the config name somewhat generic with spark.executor.resource.{resourceName}.vendor because the intention was it could be used by other cluster managers if it every applied.  So do you think its enough just to update the documentation to say its vendor-domain on k8s or would it be more obvious to change the config name to something like spark.executor.resource.{resourceName}.vendorDomain?  Another option would be I can change the name of the config to something even more generic like spark.executor.resource.{resourceName}.resourceScheme
Any opinion on what would be most obvious to users?

Reply:
vendorDomain sounds like it is a bit more explicit in terms of what it's expecting. IMO, it's expected semantic should not depend on what back-end is in play.  If vendor domain is a purely k8s component (and it may not be) then could consider scoping it `spark.kubernetes....`, but in general these are all project hydrogen related and so should not refer to any one backend.

Reply:
Agreed with @erikerlandson. The current option name sounds fine to me as long as we document clearly what values we expect users to set.